File: Chapter05/FS/WhatSize/WhatSize/WhatSize/WhatSizePage.fs
namespace WhatSize

open System
open Xamarin.Forms

type WhatSizePage() as self = 
    inherit ContentPage()

    let label = Label(FontSize = Device.GetNamedSize(NamedSize.Large, typeof<Label>),
                                 HorizontalOptions = LayoutOptions.Center,
                                 VerticalOptions = LayoutOptions.Center)

    do base.Content <- label

    do self.SizeChanged.Add(fun _ -> label.Text <- String.Format("{0} \u00D7 {1}",
                                                                 self.Width,
                                                                 self.Height))

File: Chapter05/WhatSize/WhatSize/WhatSize/WhatSizePage.cs
using System;
using Xamarin.Forms;

namespace WhatSize
{
    public class WhatSizePage : ContentPage
    {
        Label label;

        public WhatSizePage()
        {
            label = new Label
            {
                FontSize = Device.GetNamedSize(NamedSize.Large, typeof(Label)),
                HorizontalOptions = LayoutOptions.Center,
                VerticalOptions =  LayoutOptions.Center
            };

            Content = label;

            SizeChanged += OnPageSizeChanged;
        }

        void OnPageSizeChanged(object sender, EventArgs args)
        {
            label.Text = String.Format("{0} \u00D7 {1}", Width, Height);
        }
    }
}
